### PR TITLE
feat: APP-203 buy credits step 3

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-203?atlOrigin=eyJpIjoiMGQ4YWNlYzVkYzg1NGFkYmI0MTZkMjRmZWE0ZTg3NGQiLCJwIjoiaiJ9

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

https://deploy-preview-2419--regen-storybook.netlify.app/?path=/story/marketplace-organisms-agreepurchaseform--retirement
https://deploy-preview-2419--regen-storybook.netlify.app/?path=/story/marketplace-organisms-agreepurchaseform--no-retirement

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
@erikalogie @clevinson see testing instructions

**Comment by erikalogie** [comment]
This looks awesome! I just provided you with a mobile version that is every so slightly different (smaller paragraph text and icon), but otherwise, this looks amazing! Sorry for not providing this earlier too: https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=3979-86580&t=kJffMCY9IWXtRyzh-1

File: web-marketplace/src/components/organisms/AgreePurchaseForm/AgreePurchaseForm.Tradable.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <Card className="py-30 px-20 sm:py-50 sm:px-40 border-grey-300">
+      <Card className="p-20 border-grey-300 bg-grey-100 flex gap-10 flex-col sm:flex-row items-center sm:items-start">
+        <img
+          className="h-[100px] w-[100px]"
+          src={imgSrc}
+          alt="info with hand"
+        />
+        <div>
+          <Title className="text-base">
```
Comment (r41ph) [inline web-marketplace/src/components/organisms/AgreePurchaseForm/AgreePurchaseForm.Tradable.tsx:19]: I think this is going to be an `<h1>` without the variant prop. As there is a main heading in the page, maybe we should use a `<h2>` instead?

File: web-components/src/components/inputs/new/CheckboxLabel/CheckboxLabel.tsx [hunk 2]
```diff
@@ -28,19 +31,27 @@ const CheckboxLabel = forwardRef<HTMLButtonElement, CheckboxLabelProps>(
       className,
       error = false,
       helperText = '',
+      optional,
+      optionalClassName,
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-components/src/components/inputs/new/CheckboxLabel/CheckboxLabel.tsx:35]: `optional` is already used in most of our input components so I'd rather keep it as is to keep naming consistent and avoid having to replace it everywhere
But I can update `optionalTextClassName`, I agree `optionalClassName` could be misunderstood.

File: web-marketplace/src/components/organisms/AgreePurchaseForm/AgreePurchaseForm.Retirement.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    ),
+);
+const LocationStateField = lazy(
+  () =>
+    import(
+      'web-components/src/components/inputs/new/LocationStateField/LocationStateField'
+    ),
+);
+
+export type RetirementProps = {};
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-marketplace/src/components/organisms/AgreePurchaseForm/AgreePurchaseForm.Retirement.tsx:26]: no I think I had created it initially thinking that there will be some props but it's not the case so I'll just remove it

**Comment by blushi** [comment]
> This looks awesome! I just provided you with a mobile version that is every so slightly different (smaller paragraph text and icon), but otherwise, this looks amazing! Sorry for not providing this earlier too: https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=3979-86580&t=kJffMCY9IWXtRyzh-1

Could you have another look @erikalogie ?

**Review by r41ph — APPROVED** [review]

**Comment by erikalogie** [comment]
It looks kind of cut off at smaller viewports:

Screenshot 2024-08-07 at 1 47 32 PM

**Comment by blushi** [comment]
> It looks kind of cut off at smaller viewports

@erikalogie ready for another look

**Comment by erikalogie** [comment]
LGTM

**Pull request merged**